libfreerdp/core/connection.c

```diff
@@ -314,8 +314,18 @@ BOOL rdp_client_disconnect(rdpRdp* rdp)
 BOOL rdp_client_redirect(rdpRdp* rdp)
 {
 	BOOL status;
+	BOOL reconnect_channels = FALSE;
 	rdpSettings* settings = rdp->settings;
 
+	/* Ensure existing channels are disconnected */
+	rdpContext* context = rdp->context;
+	rdpChannels* channels = context->channels;
+	if (channels->connected)
```

Comment:
We need to know if the channels were connected in the first place, so that we can only call freerdp_channels_post_connect() if needed. What can happen otherwise it that we call it here, then it gets called again later(if redirect happens before normal call) which gives the error "CHANNEL_RC_ALREADY_CONNECTED". Once we call freerdp_channels_disconnect() then we cant tell since it resets that variable. It is possible, that rdp_client_reconnect may have the same potential problem.